RocketChat/Rocket.Chat
# [NEW] : API endpoints for Agent-Extension Association and Database changes

Description : This commit adds the required APIs and permissions to run these APIs for following :
1. Create agent-extension association. (Access to admin)
2. Get extension associated with given agent name. (Access to agent, admin and manager)
3. Delete extension of a given agent. (Access to admin)
4. List all free extensions. (Access to admin)
5. Get the list of agent-extension association. (Access to admin)

It adds necessary functions in the omnichannel-voip service and adds corresponding types.

In the database, it adds a new field |extension| to existing meteor |users| document.







## Proposed changes (including videos or screenshots)
This PR adds following APIs and necessary permissions
1. Create agent-extension association. (Access to admin)
2. Get extension associated with given agent name. (Access to agent, admin and manager)
3. Delete extension of a given agent. (Access to admin)
4. List all free extensions. (Access to admin)
5. Get the list of agent-extension association. (Access to admin)

Along with these APIs, it also adds the code in omnichannel-voip service to provide necessary data.

To search all free extensions, it uses Voip service (Which internally uses asterisk connector) to pull the information from Asterisk. It then looks up in to the database for all those agents which have extensions associated with it.

Then it finds the difference between allextensions and associatedextensions.

To get the allocation details, it gets all those agents from the data base which have extension associated with it.

Testing Notes :
1. Check that the API permission is correctly coded. 
2. check all the APIs added work properly.

## Issue(s)


## Steps to test or reproduce


## Further comments
We are using one service from another here. omnichannel-voip uses voip service.
